src/main/kotlin/study/step3/MovementConditionGenerator.kt

```diff
@@ -0,0 +1,5 @@
+package study.step3
+
+interface MovementConditionGenerator {
+    fun run(): Int
+}
```

Comment:
이렇게 하나의 추상 함수만 있는 경우([SAM](https://kotlinlang.org/docs/fun-interfaces.html#sam-conversions)) 함수형 인터페이스를 사용해주시면 람다식으로 쉽게 표현할 수 있어요🙂
```suggestion
fun interface MovementConditionGenerator {
    fun run(): Int
}

fun main() {
    val oneGenerator = MovementConditionGenerator { 1 }
}
```

Reply:
@saintbeller96 
아하 이전에 스터디하면서 한 번 지나가듯 본 적이 있었는데
적용예시를보니코드가 훨씬 깔끔해질 수 있네요~!